### PR TITLE
Refactor PARK_AWARD to use strong enum

### PR DESCRIPTION
**Pull request by gsckoco**
Based on #12417
Created new strong enum type `ParkAward : uint8_t`

### CODE REVIEW

File: src/openrct2/management/Award.cpp [hunk 1]
```diff
@@ -77,11 +77,11 @@ bool award_is_positive(int32_t type)
 /** More than 1/16 of the total guests must be thinking untidy thoughts. */
 static bool award_is_deserved_most_untidy(int32_t activeAwardTypes)
 {
-    if (activeAwardTypes & (1 << PARK_AWARD_MOST_BEAUTIFUL))
+    if (activeAwardTypes & (1 << static_cast<int>(ParkAward::MostBeautiful)))
```
Comment (duncanspumpkin) [inline src/openrct2/management/Award.cpp:80]: I created a little utility function for this EnumsToFlags (EnumToFlag) it means you don't need the static cast and 1 <<.

Comment (gsckoco) [reply src/openrct2/management/Award.cpp:80]: Oh okay thats sweet, I'll change that now

Comment (gsckoco) [reply src/openrct2/management/Award.cpp:80]: Where is EnumToFlag? I can't seem to find it anywhere?

Comment (tupaschoal) [reply src/openrct2/management/Award.cpp:80]: I think it's on the yet to be merged #12833

Comment (Lukasz-Pe) [reply src/openrct2/management/Award.cpp:80]: Yes, this PR is waiting for one more ok

Comment (gsckoco) [reply src/openrct2/management/Award.cpp:80]: I have now switched `static_cast<int>` to `EnumToFlag`

File: src/openrct2/management/Award.h [hunk 2]
```diff
@@ -17,26 +17,26 @@ struct Award
...
-    PARK_AWARD_MOST_DISAPPOINTING,
-    PARK_AWARD_BEST_WATER_RIDES,
-    PARK_AWARD_BEST_CUSTOM_DESIGNED_RIDES,
-    PARK_AWARD_MOST_DAZZLING_RIDE_COLOURS,
-    PARK_AWARD_MOST_CONFUSING_LAYOUT,
-    PARK_AWARD_BEST_GENTLE_RIDES,
-    PARK_AWARD_COUNT
+    MostUntidy,
+    MostTidy,
+    BestRollercoasters, // BestRollercoasters or BestRollerCoasters?
```
Comment (Gymnasiast) [inline src/openrct2/management/Award.h:24]: `BestRollerCoasters`

**Review by Gymnasiast — APPROVED** [review]

**Pull request merged**